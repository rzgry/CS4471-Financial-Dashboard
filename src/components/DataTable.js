import React from 'react';
import { Table } from 'semantic-ui-react';

const DataTable = ({ headers, data }) => (
  <Table basic="very">
    <Table.Header>
      <Table.Row>
        {headers.map(header => (
          <Table.HeaderCell key={header}>{header}</Table.HeaderCell>
        ))}
      </Table.Row>
    </Table.Header>
    <Table.Body>
      {data.map(row => (
        <Table.Row>
          {headers.map(header => (
            <Table.Cell key={header}>{row[header]}</Table.Cell>
          ))}
        </Table.Row>
      ))}
    </Table.Body>
  </Table>
);

DataTable.defaultProps = {
  headers: [],
  data: [],
};

export default DataTable;
