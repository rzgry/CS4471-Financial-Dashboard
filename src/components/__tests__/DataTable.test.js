import React from 'react';
import renderer from 'react-test-renderer';
import DataTable from '../DataTable';

describe('<DataTable />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <DataTable headers={['testA', 'testB']} data={[{ testA: 'testA', testB: 'testB' }]} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
