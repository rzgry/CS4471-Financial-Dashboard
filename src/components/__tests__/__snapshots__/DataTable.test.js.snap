// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable /> renders correctly 1`] = `
<table
  className="ui very basic table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        testA
      </th>
      <th
        className=""
      >
        testB
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
    >
      <td
        className=""
      >
        testA
      </td>
      <td
        className=""
      >
        testB
      </td>
    </tr>
  </tbody>
</table>
`;
